chore: update flarum/core references to flarum/monorepo

askvortsov1 opened the pull request.

davwheat committed:
feat: make markdown toolbar extensible (#33)

flarum-bot committed:
Bundled output for commit 1738957af7bd37e8b5f53d75b3b6858f3ca1d45d

Includes transpiled JS/TS.

[skip ci]

davwheat committed:
chore: 1.1.1 changelog

SychO9 committed:
fix: `Until reply` renaming permission of discussions broken in php 8 (#3243)

* test: `until reply` rename discussion ability
* fix: `Until reply` renaming of discussions broken in php 8

askvortsov1 committed:
Fix branch alias

askvortsov1 committed:
Use v1.0.0 of backend testing properly (#3247)

askvortsov1 committed:
Apply flarum-cli audit fix (#3248)

askvortsov1 committed:
fix: broken post/discussion soft delete (#3249)

* FIx broken post/discussion soft delete

Before the Model typescript rewrite, `pushAttributes` supported including relationship objects, which is hacky but incorrect behavior. With the rewrite, this functionality was broken.

This PR deprecates the functionality, adds a deprecated BC layer with a debug warning, and removes instances of incorrect usage.

* Update js/src/common/Model.ts

Co-authored-by: David Wheatley <hi@davwheat.dev>

* Update js/src/common/Model.ts

Co-authored-by: David Wheatley <hi@davwheat.dev>

* chore: format

Co-authored-by: David Wheatley <hi@davwheat.dev>

davwheat committed:
docs: add missing 1.1.1 changelog

This was present on the 1.1.1 branch, but not `master`.

askvortsov1 committed:
Run flarum-cli audit infra

rafaucau committed:
Rewrite extension, call API manually, take advantage of features not supported by the current library (#24)

This PR updates the entire extension. It will not use a library that has not been updated for years, so it should work with newer versions of PHP.
Additionally, it sends more information to the Akismet API to make spam detection more accurate.
The Akismet class can be used by other extensions.

Todo:
- [x] Convert frontend to TypeScript
- [x] Call Akismet API manually
- [x] Option to remove blatant spam
- [x] Permission to bypass Akismet
- [x] Sending additional parameters like `is_test`

Nice to have, but can be left for another PR:
- [ ] Suspend obvious spamers
- [ ] Send  `blog_lang` parameter
- [ ] Checking post edits

Sponsored by [forum.android.com.pl](https://forum.android.com.pl/)

StyleCIBot committed:
Apply fixes from StyleCI

flarum-bot committed:
Bundled output for commit e5ba4f4320064e9d8051a47264ec86bc25c1a5da

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
Fix `echo` statements not showing up in debug modals or console trace

This was accidentially introduced in the recent Application permissions refactor, where `formattedError`was changed to always be a string array.

askvortsov1 committed:
Format

askvortsov1 committed:
Use proper log level depending on debug mode

askvortsov1 committed:
LogDriver should work in debug mode

StyleCIBot committed:
Apply fixes from StyleCI

[ci skip] [skip ci]

rafaucau committed:
fix: duplicate dependencies in package.json (#25)

askvortsov1 committed:
Rerun audit infra --fix

askvortsov1 committed:
Switch to yarn

imorland committed:
chore: convert flag items to ItemList (#39)

* Convert to ItemList, update import namespaces

* Bump deps, add prettier

* Format js

flarum-bot committed:
Bundled output for commit 1bd32bb4e09041d38fcb3533087a0a367937570d

Includes transpiled JS/TS.

[skip ci]

imorland committed:
Update JS imports (#40)

flarum-bot committed:
Bundled output for commit 9d7f9eb0e5a4762dff632054ed5c6fa7eda4f22f

Includes transpiled JS/TS.

[skip ci]

imorland committed:
Bump JS deps

imorland committed:
Update JS imports

imorland committed:
Format JS

flarum-bot committed:
Bundled output for commit 30b2fcce665332909bf5da6be53e0f823b649219

Includes transpiled JS/TS.

[skip ci]

imorland committed:
[chore] Update JS imports (#80)

flarum-bot committed:
Bundled output for commit 0d9c95b0960c5a260c569de43bdc39682a8c56de

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
fix: Ensure `user` not false in post component

If no information about a relation exists, it will return false, not undefined. Therefore, the nullsafe operator is not enough to avoid issues here.

askvortsov1 committed:
fix: don't spellcheck custom LESS

This should stop browsers from "correcting" code, and from converting regular quotes to "smart" quotes.

askvortsov1 committed:
fix: Allow outside click on modal focus trap

Without this, alerts triggered while a modal is open can't be dismissed.

askvortsov1 committed:
Fix frontend branch name

flarum-bot committed:
Bundled output for commit 03aa76cacef96af682b18ec762f5f849ce550a79

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

flarum-bot committed:
Bundled output for commit 268d44e89dd5bfd28f69ceb9f4a76557d17e87ed

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

imorland committed:
chore: repo maintenance (#5)

* Bump deps

* Update imports

* Format

imorland committed:
chore: repo maintenance (#43)

flarum-bot committed:
Bundled output for commit d593d3ca159278bfea6721589ead09369d0f2fca

Includes transpiled JS/TS.

[skip ci]

imorland committed:
chore: repo maintenance (#30)

* Bump deps

* Update imports

* Format

flarum-bot committed:
Bundled output for commit f442a43205eacdfea6ce1c1017b6e0634aa1b070

Includes transpiled JS/TS.

[skip ci]

imorland committed:
chore: repo maintenance (#30)

* Bump deps

* Update imports

* Format

flarum-bot committed:
Bundled output for commit 542fac888d3aa5eedc9d437381caf101e8e4c1a5

Includes transpiled JS/TS.

[skip ci]

imorland committed:
chore: repo maintenance (#12)

* Bump deps

* Update imports

* Format

imorland committed:
chore: repo maintenance (#28)

* Bump deps

* Update imports

* format

flarum-bot committed:
Bundled output for commit 7b621df353d1ece1a2880ba74d68eb1ba4240605

Includes transpiled JS/TS.

[skip ci]

flarum-bot committed:
Bundled output for commit 200d48820996207938b3732cd87336d1b8648c2d

Includes transpiled JS/TS.

[skip ci]

imorland committed:
chore: repo maintenance (#32)

* Bump deps

* Update imports

imorland committed:
chore: repo maintenance (#16)

* Bump deps

* Update namespaces

* Format

flarum-bot committed:
Bundled output for commit aba58f453a323cd81d86dd36ce064ab44da0a35a

Includes transpiled JS/TS.

[skip ci]

flarum-bot committed:
Bundled output for commit b07c98b5d983cfdd911386ed46ce313002ed7db7

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
fix: return user when creating flags (#42)

Fixes https://github.com/flarum/QualityAssurance/issues/41

When the API response is ingested by `pushObject`, since the flag instance is new, there is nothing to merge with, and therefore, there's no user on the flag instance in the store. As a result, the flag is displayed with "deleted" as the creating user until a page refresh occurs.

More broadly, this could be fixed by either:
- Refactoring the JSON API layer so that all singular operations return the same set of includes.
- Merging the API Paylod with sent data on model save (for new instances only).
But the former would require a major breaking rewrite, and the latter could introduce potentially unwanted side effects, and is effectively a bandaid over inconsistent backend behavior.

askvortsov1 committed:
Some fixes to tag metadata calculations (#154)

* If a discussion doesn't exist, it can't be the last posted discussion

This fixes some rare errors where deleting the only post in the last posted discussion of a tag triggers a DB integrity exception.

* Decrement tag's discussion count when deleting a post deletes an unhidden discussion.

Co-authored-by: Sami Mazouz <sychocouldy@gmail.com>

davwheat committed:
fix: mark `near` as optional in discussion route (#3257)

flarum-bot committed:
Bundled output for commit 59a5871b59bded4019efdaca1beac7f04c8fa36e

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

askvortsov1 committed:
chore: bump @askvortsov/rich-icu-message-formatter (#3264)

Fixes https://github.com/flarum/core/issues/3259

flarum-bot committed:
Bundled output for commit c04f4c05f72962069a1db4fea3857f1aa561e67d

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

davwheat committed:
chore: switch to Webpack 5 (#35)

flarum-bot committed:
Bundled output for commit 3d864fd413a0ed3dff1a0b75009fd8267b2fe595

Includes transpiled JS/TS.

[skip ci]

rafaucau committed:
Delete package-lock.json (#26)

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit 05246fb155cd43dba1e71caa5d481c84509c00e4

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: fl-dev audit infra --fix

askvortsov1 committed:
chore: remove unnecessary tsconfig

askvortsov1 committed:
Merge branch 'master' of github.com:flarum/approval

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit 5bb0d6cb43505b5355ec88c20674f0bce36525ef

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

flarum-bot committed:
Bundled output for commit 99033734e2d815cc625a3a61d363d5c52857856a

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit 3858ebdd8617718a0a6e0416ca64218cbf4e277b

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit f5c1a0282d65624be9343f290cfffa883afa0cd4

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit 2c0e6798f96a0e24f5ae0357e604a6e30ef1e5bc

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit e9feb0aa58005e5800438cf066851f2c05cc95ff

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit 5f43267a400effede244aa7ceeae5d643773c0c1

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit 5c6f17d5086447023a38f848de7f79d80922d3e8

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit de4a0db0eca4541b5aca362546fcce524db4e4d6

Includes transpiled JS/TS.

[skip ci]

flarum-bot committed:
Bundled output for commit e03273e3686928418c8cef8e6b5f8515a4d7bcaf

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

flarum-bot committed:
Bundled output for commit b363cd18c2eb6c8039ba861539b5da1db39ed217

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
fix: typings generation script

flarum-bot committed:
Bundled output for commit 9ba882619c50652ca5566442d8d9d44b37400873

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

askvortsov1 committed:
fix: typings generation script

askvortsov1 committed:
fix: typings generation script

flarum-bot committed:
Bundled output for commit 85e6076553c504f56e67d948748bedc6660b847f

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

askvortsov1 committed:
fix: build typings script

askvortsov1 committed:
fix: build typings script

askvortsov1 committed:
fix: build typings script

askvortsov1 committed:
Merge branch 'master' of github.com:flarum/pusher

askvortsov1 committed:
Merge branch 'master' of github.com:flarum/akismet

askvortsov1 committed:
chore: bump rich icu message formatter

flarum-bot committed:
Bundled output for commit 46a7b3419b50d28b2cc3700b29824262af6622a3

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
Release v1.2

luceos committed:
fixing error in tests for missing timezone

luceos committed:
update changelog

luceos committed:
Update Application.php

update constant for next version

askvortsov1 committed:
chore: bump rich text ICU Message formatter

This fixes https://discuss.flarum.org/d/29914-utf-encoding-error-in-title/6

flarum-bot committed:
Bundled output for commit a0c5bc08280e1bd503a9fa1ac2762137ed31b9ac

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
fix: allow path repos

Temporarily configure cwd while running composer commands so that path repositories can be resolved properly.

Fixes https://github.com/flarum/package-manager/issues/6

askvortsov1 committed:
chore: bump yarn lock

askvortsov1 committed:
chore: exclude PHP 8.1 tests

These prompt deprecation warnings that break CI. We need to fix those, but that's a problem for a later time.

askvortsov1 committed:
fix: suppress xdebug warnings

askvortsov1 committed:
Merge pull request #12 from flarum/as/monorepo-prep

Monorepo Prep

askvortsov1 committed:
chore: add scaffolding exclude config

This ensures `flarum-cli audit infra --fix` works.

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: convert to TypeScript

flarum-bot committed:
Bundled output for commit f3d0bdce47cc27f96e69b9c44197dd0acccd2220

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

askvortsov1 committed:
fix: minor TS improvements

askvortsov1 committed:
chore: update yarn lockfile

flarum-bot committed:
Bundled output for commit 6d3da9c5f2f18277f439a751fb96e5a80e7b1205

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: drop unnecessary `ci` script

askvortsov1 committed:
chore: regenerate yarn.lock

flarum-bot committed:
Bundled output for commit 07fd988dbd0c116eb98144ffd845644121ae4f07

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

rafaucau committed:
feat: use `ColorPreviewInput` in the `EditTagModal` (#157)

flarum-bot committed:
Bundled output for commit eed25d18cb55724520ef172d779617188aa7b6c6

Includes transpiled JS/TS.

[skip ci]

davwheat committed:
chore: remove call to `console.log`

flarum-bot committed:
Bundled output for commit 2c0c5c01a27ab3d2fb319c8951bde46b6fad7820

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

davwheat committed:
fix: composer header overlap with controls

davwheat committed:
fix: overlap in minimized composer

luceos committed:
fix: missing import for post filterer

zerosonesfun committed:
fix(tag label): tag text misaligned in iOS Safari (#160)

askvortsov1 committed:
fix: regenerate lockfile

This should fix some JS build issues.

askvortsov1 committed:
fix: regenerate lockfile

This should fix some build issues.

askvortsov1 committed:
fix: regenerate lockfile

This should help fix some build issues.

askvortsov1 committed:
chore: make translator / rich translator typings available to exts

davwheat committed:
fix: sort posts by number instead of creation datetime (#3282)

Co-authored-by: Alexander Skvortsov <38059171+askvortsov1@users.noreply.github.com>

flarum-bot committed:
Bundled output for commit 397e4786da109282048bc3a8be5eb81be02ab0e4

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

askvortsov1 committed:
fix: refer to bundled dist-typing types instead of src ones

This ensures that extensions referring to core's built dist typings don't try to pull from core's src code.

flarum-bot committed:
Bundled output for commit 351410677fc8b94d0380bf7be015b939947941d2

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

clarkwinkelmann committed:
Make frontend use gate output instead of hard-coded logic
Hide nickname from edit modal if user cannot edit

clarkwinkelmann committed:
Fix permission check in edit modal
Previously was checking ability to edit own user instead of user selected in modal

clarkwinkelmann committed:
Fix active search jumping to last item during loading

davwheat committed:
fix: don't submit form when search clear button clicked (#3260)

flarum-bot committed:
Bundled output for commit 567c0a0d7a35469dc3faf44f1c8a9e130ddc8c6f

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

clarkwinkelmann committed:
Fix pinned pane broken after discussion creation
The pane would appear but the page content would not shift to the right because the hasPane className would be missing

flarum-bot committed:
Bundled output for commit dba6ad9d571c644dfae9ea4e7396e7f2ebc0ba80

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

flarum-bot committed:
Bundled output for commit 05e936bbe4220e46789a21f40897a0a89fe3da7a

Includes transpiled JS/TS.

[skip ci]

imorland committed:
[chore] update deprecated event comments (#41)

askvortsov1 committed:
feat: add compiled dist-typings folder

imorland committed:
fix: unread filter gambit does not return results when > 999 read discussions for a user (#3295)

therealsujitk committed:
Add toolbar support for admin panel (#36)

flarum-bot committed:
Bundled output for commit 16d5cc11e3aee5c94aeed877987cdb199a2a0d2c

Includes transpiled JS/TS.

[skip ci]

dsevillamartin committed:
Replace ColorPreviewInput preview box with 'color' input (#3271)

flarum-bot committed:
Bundled output for commit 904deda6b9cdfb2d9527c2b22f0e589b4f9bd5e8

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

dsevillamartin committed:
Fix tag page mismatches if accents/capitalizations differ in slug (#161)

Fixes flarum/core#3286

flarum-bot committed:
Bundled output for commit 671b3f4e54176edc2b8d36321d75e6d135d114be

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
Merge remote-tracking branch 'js_packages_webpack-config/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'js_packages_tsconfig/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'php_packages_testing/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'php_packages_phpstan/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_akismet/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_approval/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_bbcode/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_embed/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_emoji/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_flags/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_lang_english/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_likes/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_lock/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_markdown/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_mentions/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_nicknames/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_package_manager/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_pusher/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_statistics/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_sticky/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_subscriptions/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_suspend/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'extensions_tags/REWRITE'

askvortsov1 committed:
Merge remote-tracking branch 'framework_core/REWRITE'

askvortsov1 committed:
chore: remove centralizable repo config

- Remove .github conf folder from Flarum packages

- Remove styleci config from all packages

askvortsov1 committed:
chore: add monorepo config file

askvortsov1 committed:
chore: set up composer path repos

askvortsov1 committed:
chore: flarum-cli audit infra --fix

askvortsov1 committed:
chore: restore issue templates

askvortsov1 committed:
chore: move core's README to framework root.

dependabot[bot] committed:
chore(deps): bump glob-parent from 3.1.0 to 5.1.2 in /extensions/emoji/js (#3345)

* chore(deps): bump glob-parent in /extensions/emoji/js

Bumps [glob-parent](https://github.com/gulpjs/glob-parent) from 3.1.0 to 5.1.2.
- [Release notes](https://github.com/gulpjs/glob-parent/releases)
- [Changelog](https://github.com/gulpjs/glob-parent/blob/main/CHANGELOG.md)
- [Commits](https://github.com/gulpjs/glob-parent/compare/v3.1.0...v5.1.2)

---
updated-dependencies:
- dependency-name: glob-parent
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>

* Apply fixes from StyleCI

[ci skip] [skip ci]

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: StyleCI Bot <bot@styleci.io>

StyleCIBot committed:
Apply fixes from StyleCI

[ci skip] [skip ci]

askvortsov1 committed:
chore: add version to core/ext package.json

This is necessary for yarn workspaces to work.

askvortsov1 committed:
chore: use yarn v1 in extensions/core

askvortsov1 committed:
feat: use yarn workspaces

flarum-bot committed:
Bundled output for commit 5e82ded8c38f15e69c01a639d28917fe9bb29f01

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
fix: Assorted Typing Fixes (#3348)

With all the commits below, we resolve all outstanding typing issues in the repo, and CI jobs run green.

* fix: Convert DashboardPage and DashboardWidget to TypeScript

* fix: fix type errors in package manager ext

* fix: Convert Post component to TypeScript

* fix: avatar typings should accept null user

* fix: convert Notification component to TypeScript

* fix: properly use `typeof` in ForumApplication

* feat: make Notification content attr generic

* chore: format Notification component

* fix: Convert DiscussionRenamedNotification to TypeScript

* fix(pusher) move shims to a location where they get applied

* fix(pusher): fix some typing errors

* fix(akismet): fix some typing issues

* chore: update core dist typings

* chore(pusher): format

* fix: anchorScroll should accept string selectors

* fix: more accurately represent ApiQueryParamsPlural

* fix: convert PostStreamState to TypeScript

* chore(core): rebuild typings

* feat: allow extending app.routes

* fix: more flexible typings for highlight.ts

* fix: use primitive `number` type for Discussion typings

* fix: convert DiscussionListItem to TypeScript

* chore: rebuild core typings

* fix: final pusher type fixes

* feat: start tags TypeScript conversion

* fix: require-dev tags in pusher for CI TypeScript purposes.

* chore(core): format

* chore(tags): build dist typings

* feat(pusher): use dist types from tags.

* feat: convert flags to TypeScript

* chore(flags): generate dist typings

* fix(akismet): last type errors

* chore: update .yarn-integrity

* chore: partially run flarum-cli audit infra --fix

The tsconfig changes from that command are ignored, since we don't yet support "replacable sections" that would let us add custom config.

* chore: use type imports

* fix: broader gitattributes

* chore: run flarum-cli audit infra --monorepo --fix

* feat: make `app.data` typings extensible

* chore(core): format

* chore: boost tags TypeScript coverage

* fix(tags): further increase type coverage.

flarum-bot committed:
Bundled output for commit a595665bfbf7820c58a3911df154ced435b6fbfd

Includes transpiled JS/TS.

[skip ci]

SychO9 committed:
fix: `Post--by-start-user` class not working (#3356)

flarum-bot committed:
Bundled output for commit 46f8cf46284754f98588646fc995fef420852a0a

Includes transpiled JS/TS.

[skip ci]

luceos committed:
possibly fixes the issue with post number calculation

StyleCIBot committed:
Apply fixes from StyleCI

[ci skip] [skip ci]

luceos committed:
fix prefix failing in tests

StyleCIBot committed:
Apply fixes from StyleCI

[ci skip] [skip ci]

luceos committed:
missed another prefix for tests

askvortsov1 committed:
fix: canonicalize comparisons in Posts ListTest

That doesn't matter since there's no default sort for posts in the backend.

davwheat committed:
a11y: use `feed` aria role for DiscussionList (#3359)

* a11y: use `feed` aria role for DiscussionList

* a11y: add aria set info roles, move feed roles to more appropriate element

flarum-bot committed:
Bundled output for commit b95b3646d3016b5163de51c465cc33a5c2eb14c1

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
chore: update flarum/core references to flarum/monorepo

askvortsov1 committed:
feat: deprecate discussion `post_number_index` field.

flarum-bot committed:
Bundled output for commit 00b91518646eeddb09635bfa76b4e92591c2595f

Includes transpiled JS/TS.

[skip ci]

davwheat commented on js-packages/tsconfig/README.md:
```diff
@@ -22,7 +22,7 @@ A baseline `tsconfig.json` is provided below that you can modify as needed. This
   // and also tells your Typescript server to read core's global typings for
   // access to `dayjs` and `$` in the global namespace.
-  "include": ["src/**/*", "../vendor/flarum/core/js/dist-typings/@types/**/*"],
+  "include": ["src/**/*", "../vendor/flarum/framework/js/dist-typings/@types/**/*"],
```
I think this should remain as `core`, since this is referring to the PHP package name which is still `flarum/core`.

davwheat committed:
feat: allow multiple children to be passed to ConfirmDocumentUnload component (#3372)

flarum-bot committed:
Bundled output for commit 803f4cc792a1cd73e6fdb829d8f20e9ccd032722

Includes transpiled JS/TS.

[skip ci]

dsevillamartin committed:
fix(subscriptions): filter followed discussions when searching (#3376)

Fixes #3375

flarum-bot committed:
Bundled output for commit cd8ed6cf66789b187873072dabc43ce873493243

Includes transpiled JS/TS.

[skip ci]

SychO9 committed:
feat: Graceful frontend extension initialization failure (#3349)

flarum-bot committed:
Bundled output for commit ca7055f5d04af64755aef159d288e9337f17e885

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

davwheat committed:
chore: don't track `.yarn-integrity` in git (#3374)

flarum-bot committed:
Bundled output for commit 7c3da27fb32db6941f6401fee8b1727d3efd3ef4

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

imorland committed:
feat: add discussion sortmap to container binding (#3377)

flarum-bot committed:
Bundled output for commit 1d27f62c15b569f4cd74a24fc38a636d34933f9f

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

davwheat committed:
fix: notification timestamp mismatch in specific configurations (#3379)

When app's PHP server's timezone is not set to UTC, which would normally be done by Flarum's bootstrapping process, this call to Carbon would return a different timestamp to other calls elsewhere in Flarum, causing issues with notifications seemingly being delivered too early/late rather than at the actual time they were triggered.

flarum-bot committed:
Bundled output for commit 9f7c5defaa1d082bda9af99dfc84d08e3e512ca5

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
fix: dont ignore optional dependencies on disabled extensions. (#3352)

There is a check in the ExtensionManager::resolveExtensionOrder function that ignores optional dependencies on extensions that don't exist in the system. This is sufficient for resolution purposes.

The filter removed in this PR would ignore optional dependencies on non-enabled extensions, so when such an extension was enabled, dependency resolution would run incorrectly.

flarum-bot committed:
Bundled output for commit 431ba304348f0f923c65183e5f8e3c03ac3da9b9

Includes transpiled JS/TS, and Typescript declaration files (typings).

[skip ci]

SychO9 committed:
fix: extension listeners added before core listeners (#3373)

flarum-bot committed:
Bundled output for commit 98e607a7041b55130cfc5d0c8fb3e23b4b64e951

Includes transpiled JS/TS.

[skip ci]

imorland committed:
fix: extensions not being able to use the renderer without context (#3382)

flarum-bot committed:
Bundled output for commit b0fa795cc936a5538af892312e97bbf763625e18

Includes transpiled JS/TS.

[skip ci]

askvortsov1 committed:
feat: update tsconfig docs
